sdk/monitor/opentelemetry-exporter-azure-monitor/package.json

```diff
@@ -35,23 +35,22 @@
     "docs": "typedoc --excludePrivate --excludeNotExported --excludeExternals --mode file --out ./dist/docs ./src"
   },
   "engines": {
-    "node": ">=8.3.0"
```

Comment:
Is there a reason this was set to 8.3.0 and not 8.0.0? Feel free to let me know if I should suppress the linter rule that enforces this here

Reply:
I don't believe so, unless @hectorhdzg knows of something that the exporter needs that requires 8.3.x?

Reply:
OpenTelemetry doesn't support node < 8.5.x actually we need to update this to >=8.5.0 
https://github.com/open-telemetry/opentelemetry-js#node-support

Reply:
I can change that to 8.5.0 and turn off the linter error that complains. I can also update the README if you'd like since I'll be there:  https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/monitor/opentelemetry-exporter-azure-monitor#prerequisites

Does that work for you @hectorhdzg  ?

Reply:
/cc @bterlson @ramya-rao-a can we update our min-bar to Node 8.5.0 across the board?

Reply:
@maorleger that would be awesome

Reply:
@xirzec I am not sure if you know this but we do not have to. Rush allows exceptions and we are already using it: https://github.com/Azure/azure-sdk-for-js/blob/cd97a786e858f3166b612ccae450082479f7a600/common/config/rush/common-versions.json#L62

Reply:
Originally I wanted to turn that rule off for `opentelemetry-exporter-azure-monitor`; however, that rule couples the required node version with the presence of any node version so turning it off means there's nothing to stop someone from removing the `engines` field entirely. 

So instead I'm providing an option called `overrideNodeVersion` to @azure/azure-sdk/ts-package-json-engine-is-present 

You can see the change in this commit: https://github.com/Azure/azure-sdk-for-js/pull/13070/commits/28984d1314978da5fdc6dd5553c25582c4fba7b2

I also cleaned up the docs (they weren't accurate), added info about that override flag, and updated `opentelemetry-exporter-azure-monitor` to specify 8.5.0 as the min node version.

Reply:
@maorleger this is great!

@hectorhdzg I wonder if `node` v8.5.0 is the minimum version supported, why not use this version for the dev dependency instead of v10?

Reply:
@deyaaeldeen do you mean @types/node package?, that is interesting I can see OpenTelemetry have dependencies on latest version of the package.
https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-api/package.json#L58

So the issue is related to the code using classes that doesn't exist in previous node.js versions and potentially causing issues when running in older versions, makes total sense to update it in my opinion, I can take care of this update.

Reply:
@xirzec if you're asking from a policy perspective, I think yes. I'd be fine with 8.17 being a min bar even, I think.

Reply:
@hectorhdzg yeah I think the node version(s) specified in `engines` should match that of `@types/node`. If using the latest node version is necessary, then having that as the version in engines and `@types/node` should be fine. @xirzec @bterlson correct me if I am wrong about this. 

P.S. If you were to change the version of `@types/node`, you also need to change the version in `common/config/rush/common-versions.json`.